pyfstat/helper_functions.py

```diff
@@ -246,59 +247,55 @@ def get_comb_values(F0, frequencies, twoF, period, N=4):
     return comb_frequencies, twoF[comb_idxs], freq_err * np.ones(len(comb_idxs))
 
 
-def run_commandline(cl, raise_error=True, return_output=True):
-    """Run a string cmd as a subprocess, check for errors and return output.
+def run_commandline(
+    cl: str, raise_error: bool = True, return_output: bool = False
+) -> Union[subprocess.CompletedProcess, None]:
+    """Run a string command as a subprocess.
 
     Parameters
     ----------
-    cl: str
+    cl:
         Command to run
-    log_level: int
-        Sets the logging level for some of this function's messages.
-        See https://docs.python.org/library/logging.html#logging-levels
-        Default is '20' (INFO).
-        FIXME: Not used for all messages.
-    raise_error: bool
+    raise_error:
         If True, raise an error if the subprocess fails.
-        If False, continue and just return `0`.
-    return_output: bool
-        If True, return the captured output of the subprocess (stdout and stderr).
-        If False, return nothing on successful execution.
+        If False, just log the error, continue, and return ``None``.
+    return_output:
+        If True, return the ``subprocess.CompletedProcess`` object.
+        If False, return ``None``.
 
     Returns
     ----------
-    out: str or int, optional
-        The captured output of the subprocess (stdout and stderr)
-        if `return_output=True`.
-        0 on failed execution if `raise_error=False`.
+    out:
+        The ```subprocess.CompletedProcess`` of the subprocess
+        if ``return_output=True``. ``None`` if ``return_output=False``
+        or on  failed execution if ``raise_error=False``.
     """
 
     logger.info("Now executing: " + cl)
     if "|" in cl:
         logger.warning(
-            "Pipe ('|') found in commandline, errors may not be" " properly caught!"
+            "Pipe ('|') found in commandline, errors may not be  properly caught!"
         )
     try:
+        completed_process = subprocess.run(
+            cl,
+            check=True,
+            shell=True,
+            capture_output=True,
+            text=True,
+        )
+        if msg := completed_process.stdout:
+            logger.info(msg)
+        if msg := completed_process.stderr:
+            logger.error(msg)
         if return_output:
-            out = subprocess.check_output(
-                cl,  # what to run
-                stderr=subprocess.STDOUT,  # catch errors
-                shell=True,  # proper environment etc
-                universal_newlines=True,  # properly display linebreaks in error/output printing
-            )
-        else:
-            subprocess.check_call(cl, shell=True)
+            return completed_process
     except subprocess.CalledProcessError as e:
-        logger.error("Execution failed: {}".format(e))
-        if e.output:
-            logger.error(e.output)
+        logger.error(f"Execution failed: {e.output}")
```

Comment:
Keeping `capture_output=True` intact above, this also seems to properly show XLAL errors:
```suggestion
        logger.error(f"Execution failed: {e}")
        logger.error(e.stderr)
        logger.error(e.output)
```

Reply:
```suggestion
        logger.error(f"Execution failed: {e}")
        for label in ["stderr", "output"]:
            if msg := getattr(e, label, None):
                logger.error(msg)
```

Can we assume they always exist?

Reply:
No idea, so yeah better be safe.